app/models/pawn.rb

```diff
@@ -20,6 +20,49 @@ def promote!(new_type)
 
   private
 
+  # En Passant Logic:
+  # 1)target pawn just moved 2 space forward
+  # 2)only capture within the next turn
+  # 3)can only capture by opponent's pawn
+  # 4)opponent pawn must be on the left or right of target piece's y axis
+
+  # En Passant Capture logic
+  def en_passant_capture?(x, y)
+    return true if en_passant_possible?(x, y) || en_passant_y_diff?(x, y)
+  end
+
+  # Return false if victim piece is nil or same color
+  # Return true  if victim piece's y axis difference is equal to 1
+  def en_passant_y_diff?(x, y)
+    victim = last_moved_piece
+    return false if victim.nil? || victim.color == color
+    return true if  x == victim.x_position &&
+                    (y == victim.y_position - 1 || y == victim.y_position + 1)
+  end
+
+  # Return false if last move piece is nil (Start of game)
+  # Return false last moved piece is not pawn or its y axis difference is !2
+  # Check both both adjacent side of pawn
+  # If any have valid En Passant pawn piece, return true
+  def en_passant_possible?(x, y)
+    return false if last_moved_piece.nil?
+    return false unless last_moved_piece.y_distance(y - 2) == 2 &&
+                        type == 'Pawn'
+    [valid_en_passant_pawn?(x, y, true), valid_en_passant_pawn?(x, y, false)]
+      .include?(true)
+  end
+
+  # Check for any adjacent piece
+  # Return false if none if found
+  # Return true if opponent's pawn is found
+  def valid_en_passant_pawn?(x, y, adjacent)
+    adjacent_piece = occupant_piece(x + (adjacent ? 1 : -1), y)
+    return false if adjacent_piece.nil?
```

Comment:
This method checks if there is any valid pawn adjacent to the victim pawn(include? true or false). If any of adjacent side have a valid pawn, it will return true.

This was the rafactored version of following methods:
```Ruby
def valid_en_passant_pawn?(x, y)
return false unless valid_passant_pawn_left?(x, y)
return false unless valid_passant_pawn_right?(x. y)
false
end

def valid_passant_pawn_left?(x, y)
left_piece = find_piece(x - 1, y)
return false if left_piece.nil?
return true if left_piece.type == 'Pawn' && side_piece.color != color
false
end

def valid_passant_pawn_right?(x, y)
right_piece = find_piece(x + 1, y)
return false if right_piece.nil?
return true if right_piece.type == 'Pawn' && side_piece.color != color
false
end
```